networkx/algorithms/tests/test_dag.py

```diff
@@ -618,8 +618,25 @@ def test_is_aperiodic_selfloop():
     assert nx.is_aperiodic(G)
 
 
+def test_is_aperiodic_null_graph_raises():
+    G = nx.DiGraph()
+    pytest.raises(nx.NetworkXPointlessConcept, nx.is_aperiodic, G)
+
+
 def test_is_aperiodic_undirected_raises():
-    G = nx.Graph()
+    G = nx.Graph([(1, 2), (2, 3), (3, 1)])
+    pytest.raises(nx.NetworkXError, nx.is_aperiodic, G)
+
+
+def test_is_aperiodic_disconnected_raises():
+    G = nx.DiGraph()
+    nx.add_cycle(G, [0, 1, 2])
+    G.add_edge(3, 3)
+    pytest.raises(nx.NetworkXError, nx.is_aperiodic, G)
+
+
+def test_is_aperiodic_weakly_connected_raises():
+    G = nx.Graph([(1, 2), (2, 3)])
```

Comment:
This currently raises because the graph is undirected, but the error type is the same (which was up for discussion in the larger PR, but let's still fix this for now).
```suggestion
    G = nx.DiGraph([(1, 2), (2, 3)])
```
Let's add a `match` regex to the `pytest.raises` calls where such ambiguities are possible.